contrib/storage-http/src/main/java/org/apache/drill/exec/store/http/paginator/PagePaginator.java

```diff
@@ -64,16 +60,16 @@ public PagePaginator(Builder builder, int limit, int pageSize, String pageParam,
   }
 
   @Override
-  public String next() {
-    if (hasLimit) {
-      return super.next();
-    } else {
-      return generateNextUrl();
-    }
+  public boolean hasNext() {
+    return !partialPageReceived && (limit < 0 || (currentPage-1) * pageSize < limit);
```

Comment:
Limit code is off. A limit of 0 should be allowed: it means schema only. Fixed in the EVF V2/Easy Format Plugin PR.

Reply:
@paul-rogers I can't see that LIMIT 0 is disallowed.  This `hasNext()` method above returns false immediately in this case and 0 records are returned for the query, is that not correct?